code/controllers/subsystem/bluespace_exploration.dm

```diff
@@ -0,0 +1,483 @@
+
+/datum/controller/subsystem/bluespace_exploration/proc/prep_wipe(datum/data_holder/bluespace_exploration/data_holder, list/divided_turfs)
+	wipe_data_holder = data_holder
+	wiping_divided_turfs = divided_turfs
+	wipe_process_num = 0
+
+/datum/controller/subsystem/bluespace_exploration/proc/continue_wipe(datum/data_holder/bluespace_exploration/data_holder, list/divided_turfs, process_num)
+	var/list_element = (process_num % (CLEAR_TURF_PROCESSING_TIME/2)) + 1
+	switch(process_num)
+		if(0 to (CLEAR_TURF_PROCESSING_TIME/2)-1)
+			reset_turfs(divided_turfs[list_element])
+		if((CLEAR_TURF_PROCESSING_TIME/2) to (CLEAR_TURF_PROCESSING_TIME-1))
+			clear_turf_atoms(divided_turfs[list_element])
+		else
+			var/datum/data_holder/bluespace_exploration/data = data_holder
+			if(data.spawn_ruins)
+				addtimer(CALLBACK(src, .proc/place_ruins, data_holder), 0)
+			wiping_z_level = FALSE	//Done :)
+			return
+
+/datum/controller/subsystem/bluespace_exploration/proc/clear_turf_atoms(list/turfs)
+	//Clear atoms
+	for(var/turf/T in turfs)
+		SSair.remove_from_active(T)
+		// Remove all atoms except abstract mobs
+		var/static/list/ignored_atoms = typecacheof(list(/mob/dead, /mob/camera, /mob/dview))
+		var/list/allowed_contents = typecache_filter_list_reverse(T.GetAllContents(), ignored_atoms)
+		allowed_contents -= T
+		for(var/i in 1 to allowed_contents.len)
+			var/thing = allowed_contents[i]
+			if(ismob(thing))
+				var/mob/living/M = thing
+				if(M.key)
+					//If the mob has a key (but is DC) then teleport them to a safe z-level where they can potentially be retrieved.
+					//Since the wiping takes 90 seconds they could potentially still be on the z-level as it is wiping if they reconnect in time
+					random_teleport_atom(M)
+					to_chat(M, "<span class='warning'>You feel sick as your body lurches through space and time, the ripples of the starship that brought you here eminate no more and you get the horrible feeling that you have been left behind.</span>")
+				else
+					qdel(thing, force=TRUE)
+			else
+				qdel(thing, force=TRUE)
+
+//Randomly teleports an atom to a random z-level
+//Copy and paste of turf/open/space/transit, could probably be a global proc
+
+/datum/controller/subsystem/bluespace_exploration/proc/random_teleport_atom(atom/movable/AM)
+	set waitfor = FALSE
+	if(!AM || istype(AM, /obj/docking_port))
+		return
+	if(AM.loc != src) 	// Multi-tile objects are "in" multiple locs but its loc is it's true placement.
+		return			// Don't move multi tile objects if their origin isnt in transit
+	var/max = world.maxx-TRANSITIONEDGE
+	var/min = 1+TRANSITIONEDGE
+
+	var/list/possible_transtitons = list()
+	for(var/A in SSmapping.z_list)
+		var/datum/space_level/D = A
+		if (D.linkage == CROSSLINKED)
+			possible_transtitons += D.z_value
+	var/_z = pick(possible_transtitons)
+
+	//now select coordinates for a border turf
+	var/_x = rand(min,max)
+	var/_y = rand(min,max)
+
+	var/turf/T = locate(_x, _y, _z)
```

Comment:
What if it's a closed turf?

Reply:
This is copy and pasted from /turf/open/space/transit/proc/throw_atom(atom/movable/AM) and should probably be put on a proc inside mob itself

Reply:
Its very unlikely to be a closed turf since space is pretty empty, and if it is it doesn't matter too much since this doesn't get called much